woocommerce/woocommerce-ios
# Order details: display add-ons in separate lines instead of a comma-separated list

Closes: #10419 


## Why


We want to display add-ons in separate lines (as in `wp-admin`) instead of a comma-separated list, to improve the add-ons name/value readability.

## How

A stack view was added to `ProductDetailsTableViewCell`, which is hidden when the order item has no add-ons. It shows a vertical stack of labels with key and bolded value.

The add-ons with the same key are aggregated in `ProductDetailsCellViewModel.AddOnsViewModel` based on the design guidance pe5pgL-3AL-p2#comment-2958.

The add-ons should not affect variation attributes in order details. However, because we can't differentiate add-ons in order item metadata easily from variation attributes, attributes aren't shown when add-ons are available.

## Testing instructions


Prerequisite: the store has one order with add-ons, and one order with a variable product with attributes

- Go to the Orders tab
- Tap on an order with add-ons in one of the line items --> the add-ons should be shown in separate lines that follow the design pe5pgL-3AL-p2#comment-2958
- Go back to the Orders tab
- Tap on an order without add-ons and with variation attributes in one of the line items --> the add-ons should not be shown, and variation attributes should be shown next to the quantity and price subtitle like before

## Screenshots


order with add-ons  | order with variation attributes
-- | --
<img src="https://github.com/woocommerce/woocommerce-ios/assets/1945542/d5944005-bf7f-4b4f-92b1-5729ce69860d" width="300" /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/1945542/f67bc751-e6ff-43a8-b9f5-098074dbcda4" width="300" />


---
